content/blog/workflowsets-0-0-1/index.Rmd

```diff
@@ -0,0 +1,316 @@
+
+The input into a workflow set is a named list of preprocessors and parsnip model specifications. We'll could the [parsnip RStudio add-in](https://parsnip.tidymodels.org/reference/parsnip_addin.html) to write out the model code. Linear regression and multivariate adaptive regression spline (MARS) models will be combined with the different preprocessors. 
+
+```{r models}
+lm_spec <- linear_reg() %>% set_engine("lm")
+mars_spec <- mars() %>% set_engine("earth") %>% set_mode("regression")
+```
+
+Finally, to make the workflow set, named lists are created and given as inputs into [`workflow_set()`](https://workflowsets.tidymodels.org/reference/workflow_set.html). Note that we added the original data as-is using a "plain" R formula. 
+
+```{r lists}
+preprocessors <-
+  list(
+    plain = water ~ .,
+    pca = pca_recipe,
+    pca_rbf = pca_kernel_recipe,
+    pls = pls_recipe,
+    ica = ica_recipe,
+    mds = mds_recipe
+  )
+
+models <- list(lm = lm_spec, mars = mars_spec)
+
+meat_wflow_set <- workflow_set(preprocessors, models, cross = TRUE)
+meat_wflow_set
+```
+
+The 12 rows are the different combinations of preprocessors and models.  As will be shown below, we can evaluate them via resampling using the [`workflow_map()`](https://workflowsets.tidymodels.org/reference/workflow_map.html) function. 
+
+However, a few of these recipes have tuning parameters that are somewhat difficult to optimize. We might want to have specific tuning parameter ranges for the kernel PCA and MDS recipes. We can accomplish that in the usual way (via `parameters()` and `update()`) then update the workflow set with these options so that they are used later. 
+
+```{r param}
+mds_param <- 
+  mds_recipe %>% 
+  parameters() %>%
+  # Look at a larger neighbor size than the default range. 
+  update(neighbors = neighbors(c(20, 70)))
+
+rbf_param <- 
+  pca_kernel_recipe %>% 
+  parameters() %>%
+  # A smaller range than the default.
+  update(sigma = rbf_sigma(c(-5, 0)))
+```
+
+The [`option_add()`](https://workflowsets.tidymodels.org/reference/option_add.html) function can add these to the `option` column in the workflow set for the appropriate workflows: 
+
+```{r opt-add}
+meat_wflow_set <- 
+  meat_wflow_set %>% 
+  option_add(param_info = rbf_param, id = "pca_rbf_lm")  %>% 
+  option_add(param_info = rbf_param, id = "pca_rbf_mars") %>% 
+  option_add(param_info = mds_param, id = "mds_lm") %>% 
+  option_add(param_info = mds_param, id = "mds_mars")
+meat_wflow_set
+```
+
+Note that some of the entries in the `option` column have changed to `<opts[1]>`. 
+
+We can finally tune these models using `workflow_map()`. By default, it uses a basic grid search but any evaluation function from the `tune` and `finetune` packages can be used. We'll also pass more global options here for the resamples, grid points, and performance metric. The `seed` option resets the random number seed for each pass through the workflow set so that the results are reproducible. 
```

Comment:
```suggestion
We can finally tune these models using `workflow_map()`. By default, it uses a basic grid search but any evaluation function from the tune and finetune packages can be used. We'll also pass more global options here for the resamples, grid points, and performance metric. The `seed` option resets the random number seed for each pass through the workflow set so that the results are reproducible. 
```